lucet-runtime/lucet-runtime-internals/src/alloc/mod.rs

```diff
@@ -166,6 +183,13 @@ impl Alloc {
             return AddrLocation::Stack;
         }
 
+        let stack_emergency_start = self.slot().emergency_stack as usize;
+        let stack_emergency_end = stack_emergency_start + Slot::emergency_stack_size();
+
+        if (addr >= stack_emergency_start) && (addr < stack_emergency_end) {
+            return AddrLocation::StackEmergencyGuard;
+        }
+
```

Comment:
Can you move this to the end of the function, so that it appears in the same order that it does in the laid out memory?